### DISCUSSION TITLE
Keyball39右手トラックボール有の最下段が反応しない

### DISCUSSION THREAD

**Discussion by PippoCEO** [Yowkees/keyball#306] [Q&A]
お世話になります。
過去の質問を確認して、該当の事象が見つかりませんでした。

トラックボール有の右手側の最下段のキーが反応せず、確認ポイントを教えて欲しいです。

テスターとダイオードの配送を待っている状態です。
キースイッチ、pro microのはんだ馴染ませ直し、ダイオードの交換をしてみましたが、解決に至りませんでした。
※ダイオードは再度やり直そうと思っています

追記
テスターが届いたのでダイオード確認してみましたが、該当の箇所は通電していました。
キースイッチのテスターの当て方は、ググってみてもわかりませんでした……
pro microにROW3の印字があったので、ここかなと思いはんだを当て直してみましたが特に状況変わらず……

![IMG_5727](https://github.com/Yowkees/keyball/assets/142171155/cfc3c92f-f7f5-413b-90f1-1f4c55bbc58e)
![IMG_5729](https://github.com/Yowkees/keyball/assets/142171155/904dda29-6544-422a-9d86-e45b8a4408bc)

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。

最下段キーがすべて機能していないということで、ProMicroの半田付けミスが一番確率が高いと思います。
ご指摘の通り、ROW3の線になります。下記画像の緑点の端子ですので、これが周辺とショートしていないかをご確認ください。
また、この端子のProMicro側の半田付けや、ProMicroとミドル基板間がきちんと導通しているかもご確認いただければ幸いです。

以上、お手数ですがご確認お願いいたします。
![KB39_ROW3fix](https://github.com/Yowkees/keyball/assets/84209772/e1f39fd7-56b7-4240-b315-e1b96ebb7583)

**Reply by PippoCEO** [reply]
ご返信ありがとうございます。
教えていただいた箇所をはんだ当て直してみましたが、状況変わりませんでした。

その後、テスターで赤色をpro micro側、黒色を基盤側で当ててみたらブザーはなって導通しているようてました。
※すみません、テスター初めてのためよくわかっておりません。

周辺とショートしていないかは、どのように確認すればよいか教えてください。

**Reply by PippoCEO** [reply]
追記です。
pro microを別なものに取り付け直してピンヘッダとソケットも変えましたが、最下段だけ反応しませんでした。

**Comment by Yowkees** [comment]
ProMicroやピンソケットの交換ありがとうございます。
ミドル基板に不具合がありそうですね。

下記画像の緑点がすべて導通しているか（テスターを当ててピーと鳴るか）ご確認ください。
また、緑点とオレンジ点がショートしていない（テスターで音が鳴らない）ことも併せてご確認ください。

もし緑点どうしで導通していない組み合わせがある場合は、その間をエナメル線やゴム被覆銅線で接続していただければ修理できます。
お手数ですがご確認お願いいたします。
![KB39_ROWfix](https://github.com/Yowkees/keyball/assets/84209772/7e7c74ca-2bf9-4cb8-8f01-9f7c913098be)

**Comment by PippoCEO** [comment]
こちら解決済です。
遊舎工房様の有償技術サポートで解決いたらず、白銀ラボ様に有償修理依頼しました。
最終的には、基板ダメージか半田屑が影響している可能性があり、作り直す方が早いという方針となりお願いしました。
無事に動かすことができて大変満足しております。ありがとうございました。
